Updates to Guides > Hook section

brianduffield-okta opened the pull request:
## Description:
- **What's changed?** Updates to the Guides > Hook section, including an Overall Considerations guide (on implementing hooks in general); a revised Event Hook guide, walking through a specific example; an updated Password Import Hook guide; and a new Registration Inline Hook Guide. All three hook guides also come with an associated Glitch project.

- **Is this PR related to a Monolith release?** No

quyle-okta commented on packages/@okta/vuepress-site/docs/guides/event-hook-implementation/enable-and-verify/index.md:
```diff
@@ -0,0 +1,44 @@
+
+8. You can complete the one-time verification Okta call at this time or verify the Event Hook later. If using the Glitch example, proceed to verification.
+
+> **Note** You can also set up an Event Hook using an API. See [Inline Hooks Management](/docs/reference/api/inline-hooks/) for further information.
```
I think we want to link to [Event Hooks Management API](http://localhost:8080/docs/reference/api/event-hooks/)  here instead of Inline Hooks Management.

quyle-okta commented on packages/@okta/vuepress-site/docs/guides/event-hook-implementation/initial-verification/index.md:
```diff
@@ -0,0 +1,12 @@
+
+Add the following code to your external service to address this request.
+
+> **Note:** Also, make sure to have the required default code and packages in your project. See [Overview and Considerations](/docs/guides/hook-implementations/overview-and-considerations/hook-implementation-setup) for further information.
```
The "Overview and Considerations" link is broken.

quyle-okta commented on packages/@okta/vuepress-site/docs/guides/password-import-inline-hook/activate-and-enable/index.md:
```diff
@@ -0,0 +1,23 @@
+
+The Password Import Inline Hook is now set up with a status of active.
+
+> **Note:** You can also setup an inline hook using an API. See [Inline Hook Setup](/docs/reference/api/inline-hooks/) for further information.
```
Instead of "Inline Hook Setup", can we call this "Inline Hooks Management API" as the link text?

quyle-okta commented on packages/@okta/vuepress-site/docs/guides/registration-inline-hook/activate-and-enable/index.md:
```diff
@@ -0,0 +1,35 @@
+
+The Registration Inline Hook is now set up with a status of active.
+
+> **Note:** You can also setup an inline hook using an API. See [Inline Hook Setup](/docs/reference/api/inline-hooks/) for further information.
```
Instead of "Inline Hook Setup", can we call this "Inline Hooks Management API" as the link text?

quyle-okta commented on packages/@okta/vuepress-site/docs/guides/registration-inline-hook/test-registration-hook/index.md:
```diff
@@ -0,0 +1,28 @@
+
+To run a test of your Registration Inline Hook, go to the Okta sign-in page for your Okta org, click the "Sign Up" link, and attempt to self-register.
+
+- If you use an allowable email domain, the user registration goes through.
```
My comment here is to put in a sample email for user to use based on the glitch code so we could say something like:

If you use an allowable email domain, for example, “john@yourDomain.com”, the user registration goes through.

jakubvul-okta commented on packages/@okta/vuepress-site/docs/guides/password-import-inline-hook/activate-and-enable/index.md:
```diff
@@ -0,0 +1,23 @@
+
+The Password Import Inline Hook is now set up with a status of active.
+
+> **Note:** You can also set up an inline hook using an API. See [Inline Hooks Management API](/docs/reference/api/inline-hooks/) for further information.
```
As on the other page, we can link to the exact POST operation, right?

jakubvul-okta commented on packages/@okta/vuepress-site/docs/guides/password-import-inline-hook/get-submitted-credentials/index.md:
```diff
@@ -0,0 +1,11 @@
+
+We need to get the values of `data.credential.username` and `data.credential.password` from within the body of the Password Import Inline Hook request received from Okta. These properties contain the credentials submitted by the end user who is trying to sign in.
+
+> **Note:** Make sure to have the required default code and packages in your project. See [Overview and Considerations](/docs/guides/overview-and-considerations)--> for further information.
```
```suggestion
> **Note:** Make sure to have the required default code and packages in your project. See [Overview and considerations](/docs/guides/overview-and-considerations)--> for further information.
```

jakubvul-okta commented on packages/@okta/vuepress-site/docs/guides/password-import-inline-hook/import-users/index.md:
```diff
@@ -0,0 +1,11 @@
+title: Import Users
+---
+
+Upload some users into your Okta org using the [Password Import Inline Hook API](https://developer.okta.com/docs/reference/api/users/#create-user-with-password-import-inline-hook). These users provide data to test your inline hook.
```
```suggestion
Upload some users into your Okta org using the [Password Import Inline Hook API](/docs/reference/api/users/#create-user-with-password-import-inline-hook). These users provide data to test your Inline Hook.
```

jakubvul-okta commented on packages/@okta/vuepress-site/docs/guides/password-import-inline-hook/overview/index.md:
```diff
@@ -0,0 +1,25 @@
+
+This guide provides example code for an external service to respond to calls from a Password Import Inline Hook.
+
+In the following example, the external service code parses requests from Okta and responds to Okta with commands that indicate whether the end user's credentials are valid. If the credentials are valid, the password imports into the Okta org.
```
```suggestion
In the following example, the external service code parses requests from Okta and responds to Okta with commands that indicate whether the end user's credentials are valid. If the credentials are valid, the password is imported into the Okta org.
```

jakubvul-okta commented on packages/@okta/vuepress-site/docs/guides/password-import-inline-hook/overview/index.md:
```diff
@@ -0,0 +1,25 @@
+
+In the following example, the external service code parses requests from Okta and responds to Okta with commands that indicate whether the end user's credentials are valid. If the credentials are valid, the password imports into the Okta org.
+
+At a high-level, the following workflow occurs:
```
```suggestion
At a high level, the following workflow occurs:
```

jakubvul-okta commented on packages/@okta/vuepress-site/docs/guides/password-import-inline-hook/test-password-import-hook/index.md:
```diff
@@ -0,0 +1,16 @@
+
+To run a test of your Password Import Inline Hook, go to the Okta sign-in page for your Okta org.
+
+- Start by signing in with one of the users from the data store, for example, "michelletest@doesnotexist.com", and enter an incorrect password.
```
```suggestion
- Start by signing in with one of the users from the data store, for example, "michelletest@example.com", and enter an incorrect password.
```

jakubvul-okta commented on packages/@okta/vuepress-site/docs/guides/password-import-inline-hook/test-password-import-hook/index.md:
```diff
@@ -0,0 +1,16 @@
+- Your result should be access to the Okta org and the import of the user's password into Okta.
+- Sign out and sign in again to ensure the hook is no longer called (by reviewing the Developer Console logs).
+
+> **Tip:** Review the [troubleshooting*](/docs/guides/overview-and-considerations/troubleshooting) section for information if encountering any setup or configuration difficulties.
```
```suggestion
> **Note:** Review the [troubleshooting*](/docs/guides/overview-and-considerations/troubleshooting) section for information if encountering any setup or configuration difficulties.
```

jakubvul-okta commented on packages/@okta/vuepress-site/docs/guides/registration-inline-hook/send-response-code/nodejs/send-response.md:
```diff
@@ -0,0 +1,51 @@
+  console.log('Email for ' + request.body.data.userProfile['firstName'] + " " + request.body.data.userProfile['lastName'] + " " + request.body.data.userProfile['email']);
+
+  var emailRegistration = request.body.data.userProfile['email'];
+  if (emailRegistration.includes("yourDomain.com")){
```
```suggestion
  if (emailRegistration.includes("example.com")){
```

jakubvul-okta replied:
Please ensure that all of the example domains in this doc are `example.com`.

brianduffield-okta replied:
Yes, did a search and replace. And will do so in Glitch projects as well.

jakubvul-okta commented on packages/@okta/vuepress-site/docs/guides/registration-inline-hook/send-response-code/nodejs/send-response.md:
```diff
@@ -0,0 +1,51 @@
+For this example, only emails containing "yourDomain.com" are allowed and can self-register. You can also include an `error` object in the response for those users denied registration.
```
```suggestion
For this example, only emails containing "example.com" are allowed and can self-register. You can also include an `error` object in the response for those users denied registration.
```

jakubvul-okta commented on packages/@okta/vuepress-site/docs/guides/registration-inline-hook/activate-and-enable/index.md:
```diff
@@ -0,0 +1,35 @@
+
+The Registration Inline Hook is now set up with a status of active.
+
+> **Note:** You can also setup an inline hook using an API. See [Inline Hooks Management API](/docs/reference/api/inline-hooks/) for further information.
```
More specific URL please

jakubvul-okta commented on packages/@okta/vuepress-site/docs/guides/registration-inline-hook/add-registration-code/nodejs/get-submitted-credentials.md:
```diff
@@ -0,0 +1,14 @@
+The following code represents a simple check on the email property of an incoming request from Okta; that is, the email used to self-register. The values of `data.userProfile` from within the request body contains the email property that will be verified. The variable `emailRegistration` stores the value of the email.
```
```suggestion
The following code represents a simple check on the `email` property of an incoming request from Okta; that is, the email used to self-register. The values of `data.userProfile` from within the request body contains the `email` property that will be verified. The variable `emailRegistration` stores the value of the email.
```

Since I assume we're talking about the actual property in the object

jakubvul-okta commented on packages/@okta/vuepress-site/docs/guides/registration-inline-hook/add-registration-code/nodejs/get-submitted-credentials.md:
````diff
@@ -0,0 +1,14 @@
+}
+```
+
+>**Note:** The method definition begun in this code snippet is incomplete and is completed in the <GuideLink link="../send-response-code/">Send response</GuideLink> section.
````
```suggestion
> **Note:** The method definition begun in this code snippet is incomplete and is completed in the <GuideLink link="../send-response-code/">Send response</GuideLink> section.
```

jakubvul-okta commented on packages/@okta/vuepress-site/docs/guides/registration-inline-hook/overview/index.md:
```diff
@@ -0,0 +1,24 @@
+
+This guide provides example code for an external service to respond to calls from a Registration Inline Hook.
+
+In the following example, the external service code parses requests from Okta and responds to Okta with commands that indicate whether the end user's email domain is valid and allowed to register.
```
```suggestion
In the following example, the external service code parses requests from Okta and responds with commands that indicate whether the end user's email domain is valid and allowed to register.
```

jakubvul-okta commented on packages/@okta/vuepress-site/docs/guides/registration-inline-hook/test-registration-hook/index.md:
```diff
@@ -0,0 +1,28 @@
+- If you use an allowable email domain, such as `john@yourDomain.com`, the user registration goes through.
+- If you use an incorrect email domain, the user registration is denied; review the error message, which displays the error summary from the external service code and is passed back to Okta.
+
+> **Tip:** Review [Troubleshooting hook implementations](/docs/guides/overview-and-considerations/troubleshooting) for information if encountering any setup or configuration difficulties.
```
```suggestion
> **Note:** Review [Troubleshooting hook implementations](/docs/guides/overview-and-considerations/troubleshooting) for information if encountering any setup or configuration difficulties.
```

brianduffield-okta commented on packages/@okta/vuepress-site/docs/guides/event-hook-implementation/enable-and-verify/index.md:
```diff
@@ -0,0 +1,44 @@
+
+or
+
+- In the Event Hooks table, click the Actions drop-down menu of any **UNVERIFIED** Event Hook, and select **Verify**, again, making sure that your external service is ready for the verification call.
```
Implemented your change, but just subbed out "Once" for "After", as per Microsoft Manual of Style

brianduffield-okta commented on packages/@okta/vuepress-site/docs/guides/event-hook-implementation/event-object/nodejs/event-object.md:
```diff
@@ -0,0 +1,100 @@
+            {
+                "id": "00u3m90rxKjGQ0G6L1d6",
+                "type": "User",
+                "alternateId": "johndoet@doesnotexist.com",
```
Thanks! Will do.

brianduffield-okta commented on packages/@okta/vuepress-site/docs/guides/overview-and-considerations/setting-up-an-external-service/index.md:
```diff
@@ -0,0 +1,13 @@
+
+[Glitch](https://www.glitch.com) is a browser-based development environment that can build a full-stack web application online. You can use their template applications to implement an external service that receives the outbound calls from Okta orgs.
+
+Start with a new Node.js project built on the Express framework or a Node.js SQLite database application and use the code snippets in the following examples to implement the example hooks. Or remix &ndash; copy &ndash; the Glitch projects for each hook in the following sections to have a working code sample.
```
Yes, but it's Glitch's quirky way of saying "copy" in the UI; I removed the en-dashes and kept "remix" in parentheses as users will need to click "Remix on Glitch" to make their own copy of the project. To make it clear, I included the full phrase and will bold.

albertchen-okta reviewed (commented):
recommend deleting "Implementations" since every guide is a how-to implementation, would be best to keep the action works in the titles of the guides

brianduffield-okta commented:
Update Navigation, as per Albert

Changed "Hooks Implementations" back to "Hooks"
Changed "Overview and Considerations" to "Common Hook Set-up Steps"
Changed "Hook Implementation Setup" to "Overview"

Also added note that Preview Tab only available for Developer Edition Orgs.

brianduffield-okta commented:
@quyle-okta  Will confirm with Albert tomorrow; he did mention that the preview tab would be rolled out to Developer orgs shortly, and this "Note" was an interim doc. I was trying to differentiate between the Developer orgs and the Preview orgs; the Developer Console is available in each. But I'll double-check this terminology as well.

albertchen-okta commented:
> @quyle-okta Will confirm with Albert tomorrow; he did mention that the preview tab would be rolled out to Developer orgs shortly, and this "Note" was an interim doc. I was trying to differentiate between the Developer orgs and the Preview orgs; the Developer Console is available in each. But I'll double-check this terminology as well.

Sorry guys - I totally confused this preview with event hook preview, which is not available for dev orgs yet. In this case, the note is not required. 

> EDIT: What's the difference between the Developer Console and Okta Developer Edition Org? It's a bit confusing to me and probably will be to a developer.

orgs have edition types, and one type of org edition == Developer Edition. So a "developer org" is synonymous with a "developer edition" type org. It happens to be that the developer console UI is a FF that is hard coded to dev edition type orgs, so every dev edition type org features the dev console (today, but is going away ~march 2021 when we deprecate the dev console.

quyle-okta commented:
> @quyle-okta Will confirm with Albert tomorrow; he did mention that the preview tab would be rolled out to Developer orgs shortly, and this "Note" was an interim doc. I was trying to differentiate between the Developer orgs and the Preview orgs; the Developer Console is available in each. But I'll double-check this terminology as well.

@brianduffield-okta I think there is some confusion here between Preview for Inline Hook and Preview for Event Hook.  
Inline Hook Preview is already GA and available for SAML and Registration Inline Hooks.
Event Hook Preview is in Beta and we plan to make it available to all Developer Orgs as part of our Jan 2021 MC so probably be available in a Feb major release:
https://oktainc.atlassian.net/browse/OKTA-345126

christinechoy-okta reviewed (commented):
Provided minor feedback on the Event Hook Concept update.

quyle-okta commented on packages/@okta/vuepress-site/docs/guides/common-hook-set-up-steps/overview/index.md:
```diff
@@ -0,0 +1,17 @@
+
+A secure web server and application is a requirement to implement an Okta Event or Inline Hook, and is referred to as the external service in these guides.
+
+For instructional purposes, the following example guides for Event Hooks and Inline Hook types use the third-party site [Glitch](https://glitch.com) to function as an online external service, one you can use to quickly implement the sample code to preview the functionality of an Event or Inline Hook. You also need a working dev or preview Okta org to configure the hooks and implement the hook fun ...
```
This part of the sentence is a bit awkward:

"....function as an online external service, one you can use to quickly implement..."

Can we reword it a bit?

brianduffield-okta replied:
Thanks, agree it was a bit clunky. Split into two sentences: 

"For instructional purposes, the following guides for Event Hooks and Inline Hook types use the third-party site [Glitch](https://glitch.com), which functions as an online external service. You can use the example code to quickly implement the hook and preview the functionality. You also need a working dev or preview Okta org to configure the hooks and implement the hook functionality."

quyle-okta commented on packages/@okta/vuepress-site/docs/guides/password-import-inline-hook/test-password-import-hook/index.md:
```diff
@@ -0,0 +1,16 @@
+- Your result should be access to the Okta org and the import of the user's password into Okta.
+- Sign out and sign in again to ensure the hook is no longer called (by reviewing the Developer Console logs).
+
+> **Note:** Review the [troubleshooting*](/docs/guides/common-hook-set-up-steps/troubleshooting) section for information if encountering any setup or configuration difficulties.
```
I think there is an extra asterik character that needs to be removed from the "troubleshooting*" text link.
